JustSaying/AwsTools/MessageHandling/SnsTopicBase.cs

```diff
@@ -37,19 +37,16 @@ protected SnsTopicBase(IMessageSerialisationRegister serialisationRegister, ILog
             IMessageSubjectProvider messageSubjectProvider)
         {
             _serialisationRegister = serialisationRegister;
-            _log = loggerFactory.CreateLogger("JustSaying");
             _eventLog = loggerFactory.CreateLogger("EventLog");
             _snsWriteConfiguration = snsWriteConfiguration;
             _messageSubjectProvider = messageSubjectProvider;
         }
 
         public abstract Task<bool> ExistsAsync();
         
-        public Task PublishAsync(Message message) => PublishAsync(message, CancellationToken.None);
-
-        public async Task PublishAsync(Message message, CancellationToken cancellationToken)
+        public async Task PublishAsync(PublishEnvelope env, CancellationToken cancellationToken)
```

Comment:
Perhaps optional CT? I can imagine a lot of people rely on the timeout and won't want to pass one in.

Reply:
Covered by the extensions

Reply:
I was mulling this over in my review, but this might/could be catered for by extension method(s).